content/docs/command-reference/exp/run.md

```diff
@@ -81,16 +81,20 @@ Then, in your code either call the `dvc.api.make_checkpoint()` function
 (Python), or write a signal file (any programming language) following the same
 steps as `make_checkpoint()` — please refer to its reference for details.
 
-You can now use `dvc exp run` to begin the experiment. If the process gets
-interrupted (e.g. with `[Ctrl] C` or by an error), all the checkpoints so far
-will be preserved. When a run finishes normally, a final checkpoint will be
-added (if needed) to wrap up the experiment.
+You can now use `dvc exp run` to begin the experiment. All checkpoints
+registered at runtime will be preserved even if the process gets interrupted
+(e.g. with `[Ctrl] C`, or by an error\*). A final checkpoint will be added (if
+needed) to wrap up the experiment.
+
+> \* The stage command(s) should return a non-error exit code (`0`) for the
+> final checkpoint to happen. Otherwise, it will not be possible to continue
+> this experiment as described below (it will restart from scratch).
```

Comment:
It shouldn't start from scratch, it will just potentially start a modified experiment if there are "dirty" changes left over that happened after the last checkpoint was created.

Reply:
OK, I wasn't sure about this... (removed that last sentence for now). In #2399 you mentioned:

> step 3 will create a new experiment because there was no final cleanup checkpoint

But I didn't test what this looks exactly. What's a "new experiment" here, if it does continue from the last checkpoint?

Reply:
Take a look at https://github.com/iterative/dvc-checkpoints-mnist/pull/2. That should help illustrate what happens, which is the same thing that would happen if someone changed parameters or code before running `exp run` again. 

`(it will restart from scratch)` will be misleading here I think.

Reply:
Hm OK I see. I had to intro the concept of "wrap-up checkpoint" to explain this. See 1ceeebef